dot/network/service.go

```diff
@@ -218,12 +219,46 @@ func (s *Service) Start() error {
 		s.validateBlockAnnounceHandshake,
 		decodeBlockAnnounceMessage,
 		s.handleBlockAnnounceMessage,
+		nil,
 		false,
 	)
 	if err != nil {
 		logger.Warn("failed to register notifications protocol", "sub-protocol", blockAnnounceID, "error", err)
 	}
 
+	txnBatch := make(chan *transactionBatchMessage, batchSize)
+
+	txnBatchHandler := func(peer peer.ID, msg NotificationsMessage) (msgs []*transactionBatchMessage, err error) {
```

Comment:
I'd separate this into a separate function and put it in `transactions.go` and add some unit tests (seems like it's not covered for some reason?)

Reply:
Done. Previously we had a different handler for the test. I have updated the code to use the same handler.